src/parser/jobs/sam/modules/Sen.tsx

```diff
@@ -0,0 +1,413 @@
+		this.data.actions.HAGAKURE.id,
+	]
+
+	THINGS_WE_WANT_IN_THE_TABLE = [
+		this.data.actions.HAKAZE.id,
+		this.data.actions.JINPU.id,
+		this.data.actions.SHIFU.id,
+		this.data.actions.FUKO.id,
+
+		this.data.actions.GEKKO.id,
+		this.data.actions.KASHA.id,
+		this.data.actions.YUKIKAZE.id,
+		this.data.actions.MANGETSU.id,
+		this.data.actions.OKA.id,
+
+		this.data.actions.OGI_NAMIKIRI.id,
+		this.data.actions.KAESHI_NAMIKIRI.id,
+
+		// Sen Spenders
+		this.data.actions.HIGANBANA.id,
+		this.data.actions.TENKA_GOKEN.id,
+		this.data.actions.MIDARE_SETSUGEKKA.id,
+		this.data.actions.HAGAKURE.id,
+
+		// I'm leaving these in as they are a way to handle Filler. not the usual way, but a way
+		this.data.actions.ENPI.id,
+		this.data.actions.HISSATSU_YATEN.id,
+		this.data.actions.HISSATSU_GYOTEN.id,
+
+		// OGCDS IGAF about.
+		this.data.actions.HISSATSU_KAITEN.id,
+		this.data.actions.MEIKYO_SHISUI.id,
+	]
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		this.addEventHook(playerFilter.type('action').action(oneOf(this.THINGS_WE_WANT_IN_THE_TABLE)), this.checkCastAndPush)
+		this.addEventHook(playerFilter.type('combo').action(oneOf(this.SEN_ACTIONS)), this.onSenGen)
+		this.addEventHook(playerFilter.type('action').action(oneOf(this.SEN_REMOVERS)), this.remove)
+		this.addEventHook(filter<Event>().type('death').actor(this.parser.actor.id), this.onDeath)
+
+		// Suggestion time~
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	// Handles Sen Gen
+	private onSenGen(event: Events['combo']): void {
+		const action = this.data.getAction(event.action)
+
+		const lastSenState = this.lastSenState
+
+		if (lastSenState != null && lastSenState.end == null) { // The state already exists
+
+			switch (action) {
+			case this.data.actions.YUKIKAZE:
+				lastSenState.currentSetsu++
+
+				if (lastSenState.currentSetsu > 1) {
+					lastSenState.overwriteSetsus++
+					lastSenState.currentSetsu = 1
+					lastSenState.isOverwrite = true
+				}
+				break
+
+			case this.data.actions.GEKKO:
+			case this.data.actions.MANGETSU:
+				lastSenState.currentGetsu++
+
+				if (lastSenState.currentGetsu > 1) {
+					lastSenState.overwriteGetsus++
+					lastSenState.currentGetsu = 1
+					lastSenState.isOverwrite = true
+				}
+
+				break
+
+			case this.data.actions.KASHA:
+			case this.data.actions.OKA:
+				lastSenState.currentKa++
+
+				if (lastSenState.currentKa > 1) {
+					lastSenState.overwriteKas++
+					lastSenState.currentKa = 1
+					lastSenState.isOverwrite = true
+				}
+
+				break
+			}
+		}
+	}
+
+	// Function that handles SenState check, if no senState call the maker and then push to the rotation
+	checkCastAndPush(event: Events['action']) : void {
+		// check the sen state, if undefined/not active, make one
+
+		let lastSenState = this.lastSenState
+
+		if ((typeof lastSenState === 'undefined') || (lastSenState.isDone === true)) {
```

Comment:
```suggestion
		if ((lastSenState == null) || (lastSenState.isDone === true)) {
```
Please use == null (or != as necessary) when checking for null/undefined values.